File: src/main/java/com/面试中的算法/好未来/Main2.go
///*
//* type TreeNode struct {
//*   Val int
//*   Left *TreeNode
//*   Right *TreeNode
//* }
//*/
//
///**
//* 代码中的类名、方法名、参数名已经指定，请勿修改，直接返回方法规定的值即可
//*
//*
//* @param root TreeNode类
//* @param target int整型
//* @return int整型
//*/
//func searchBST( root *TreeNode ,  target int ) int {
//   // write code here
//
//   if root==nil{
//       return -1;
//   }
//   if(root.val==target){
//       return root.val;
//   }
//   if(root.val>target){
//       return searchBST(root.Left,target);
//   }else{
//       return searchBST(root.Right,target);
//   }
//}

File: src/main/java/com/面试中的算法/好未来/Main2.java
package com.面试中的算法.好未来;

/**
 * @author xing'chen
 * @version 1.0
 * @description: TODO
 * @date 2024/9/10 18:31
 */
public class Main2 {
    public int tribonacci (int n) {
        if(n==0){
            return 0;
        }
        if(n==1||n==2){
            return 1;
        }
        int t0=0,t1=1,t2=1;
        int tn=0;
        for(int i=3;i<=n;i++){
            tn=t0+t1+t2;
            t0=t1;
            t1=t2;
            t2=tn;
        }

        return tn;
        // write code here
    }
}

